tests/test_cached_property.py

```diff
@@ -37,7 +40,43 @@ def add_cached(self):
 
             return self.cached_total
 
-        def run_threads(self, num_threads):
+        @cached_property_decorator
+        def _protected_add_cached(self):
+            if threadsafe:
+                time.sleep(1)
```

Comment:
I had a test failure when I first ran the tests without making any changes. The threaded tests failed because (I think) one thread tears ahead and sets the cache, meaning the the other threads don't trigger the side effect of the property and use the cache instead.
So I am a bit unsure if I've caused any extra failures. I tried to fix the tests but I'm not really sure what they are trying to test so I don't know the best way to fix them.